build.sh

```diff
@@ -1,4 +1,6 @@
 #!/bin/sh
-javac -cp lib/junit-4.12.jar:lib/hamcrest-core-1.3.jar *.java
+javac -cp lib/junit-4.12.jar:lib/hamcrest-core-1.3.jar build/src/java/*.java
```

Comment:
Your script would work better if 
* java files should be moved to `src/java` and `test/java`, build is only for compiled class files
* `src/java/*.java` and `test/java/*.java` were part of the same command, otherwise the test class won't be able to see the target classes
* Once you make these changes, it will work much better if you add `build` to the classpath for the test and app runner (although cd'ing to the build/src/java directory was a great solution to the problem)